File: Natural Language/AddCommenter.cs
public static List<List<int>> Permutations(List<int> lst) 

{ 

if (lst.Count == 0) 

{ 

        return new List<List<int>>(); 

} 

else if (lst.Count == 1) 

{ 

        return new List<List<int>> { lst }; 

} 

else 

{ 

        List<List<int>> result = new List<List<int>>(); 

        for (int I = 0; I < lst.Count; i++) 

        { 

            int temp = lst[i]; 

            List<int> remaining = new List<int>(lst); 

            remaining.RemoveAt(i); 

            foreach (List<int> p in Permutations(remaining)) 

            { 

                List<int> permutation = new List<int> { temp }; 

                permutation.AddRange(p); 

                result.Add(permutation); 

            } 

        } 

        return result; 

} 

}

File: Natural Language/AddCommenter.py
def permutations(lst): 

    if len(lst) == 0: 

        return [] 

    elif len(lst) == 1: 

        return [lst] 

    else: 

        result = [] 

        for i in range(len(lst)): 

            temp = lst[i] 

            remaining = lst[:i] + lst[i+1:] 

            for p in permutations(remaining): 

                result.append([temp] + p) 

        return result

